server/src/main/java/org/elasticsearch/action/admin/indices/rollover/Condition.java

```diff
@@ -93,5 +96,22 @@ protected Result(Condition condition, boolean matched) {
             this.condition = condition;
             this.matched = matched;
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Result result = (Result) o;
+            return Objects.equals(condition, result.condition);
```

Comment:
Yes I think it should, whether a condition matched or not does not matter to identify which condition it is?

Reply:
Same here, I see now why it doesn't matter for how we use the object, but if just looking at the object alone one could think it is part of the object. I think it would also be good to add this explanation as a javadoc comment to the equals method if we leave it like this.